Creating the Objects Pipe and PipeGroup

edugabriel12 opened the pull request.

edugabriel12 committed:
Creating pipe.pl

Creating the Object Pipe.

edugabriel12 committed:
Creating pipeGroup.pl

Creating the Object PipeGroup.

edugabriel12 committed:
Update pipeGroup.pl

diego-aquino commented on Prolog/models/pipe.pl:
```diff
@@ -1,3 +1,58 @@
+
+toUpString(Pipe, toUpString):-
+	toStringLines(Pipe,StringLines),
+	list:join(StringLines, nl, toUpString).
```
Should `toUpString` start with an upper "T" (variable, not atom)?

diego-aquino commented on Prolog/models/pipe.pl:
```diff
@@ -1,3 +1,58 @@
+
+toDownString(Pipe, ToDownString):-
+	toUpString(Pipe,ToUpStringPipe),
+	list:join(List, nl, toUpStringPipe),
```
Did you mean `ToUpStringPipe`?

diego-aquino commented on Prolog/models/pipeGroup.pl:
```diff
@@ -1,3 +1,62 @@
+
+create(OriginX, OriginY, TopPipe, BottomPipe, Width, Height, HoleOriginY, HoleHeight, PipeGroup):-
+	TopPipeHeight is HoleOriginY - OriginY,
+	TopPipeHeight is HoleOriginY - OriginY,
```
Why are these lines duplicated?

diego-aquino commented on Prolog/models/pipeGroup.pl:
```diff
@@ -1,3 +1,62 @@
+	TopPipeHeight is HoleOriginY - OriginY,
+	TopPipeHeight is HoleOriginY - OriginY,
+	BottomPipeHeight is Height - TopPipeHeight - HoleHeight,
+	BottomPipeOriginY is originY + TopPipeHeight + HoleHeight,
```
`OriginY`?

diego-aquino commented on Prolog/models/pipeGroup.pl:
```diff
@@ -1,3 +1,62 @@
+	width(PipeGroup, Width),
+	pipe:toString(TopPipe, TopPipeAsString),
+	pipe:toString(BottomPipe, BottomPipeAsString),
+	list:createList(Width, " ", HoleLine),
```
This will create a list and unify it to `HoleLine`, but HoleLine should be a string. Did you forget to use join?

edugabriel12 committed:
Fixing pipe.pl

Switching lower case letters to upper case.

edugabriel12 committed:
Fixing pipeGroup.pl

- Removing duplicated variable.
- Switching lower case letters to upper case.
- Fixing HoleLine
